server/src/routes/v1/apps/handlers/createApp.js

```diff
@@ -154,13 +154,40 @@ module.exports = {
             }
 
             const organisationId = organisation.id
-            const requestUserId = currentUserId
+            let ownerUserId = currentUserId
             const developerUserId = appDeveloper.id
 
+            if (appJsonPayload.owner) {
+                const { email, name } = appJsonPayload.owner
+                const appOwner = await getUserByEmail(email, knex)
+
+                if (appOwner == null) {
+                    const ownerUser = await createUser(
+                        {
+                            email,
+                            name,
+                        },
+                        knex,
+                        trx
+                    )
+                    ownerUserId = ownerUser.id
+                    await addUserToOrganisation(
+                        {
+                            userId: ownerUserId,
+                            organisationId: organisation.id,
+                        },
+                        knex,
+                        trx
+                    )
+                } else {
+                    ownerUserId = appOwner.id
+                }
+            }
+
             //Create the basic app
             const dbApp = await createApp(
                 {
-                    userId: requestUserId,
+                    userId: ownerUserId,
```

Comment:
created by id should still be the request user id, If i'm as admin add an app, with another owner and developer, we should still be able to track that it was me that did this action

Reply:
and the same applies to the below changes